src/Transformer.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\fb_instant_articles\Transformer.
+ */
+
+namespace Drupal\fb_instant_articles;
+
+/**
+ * Encapsulates Drupal-specific logic when using the Transformer class.
+ *
+ * Class Transformer
+ * @package Drupal\fb_instant_articles
+ */
+class Transformer extends \Facebook\InstantArticles\Transformer\Transformer {
+
+  /**
+   * Transformer constructor.
+   *
+   * Adds Drupal hook to allow transformer rules to be added by other modules.
+   * We invoke the hook in a constructor - as opposed to inside transform() or a
+   * new invokeRules() method - because of the way Transformer is normally used.
+   *
+   * @see hook_fb_instant_articles_transformer_rules()
+   *
+   * Note the parent class does not have a constructor, so we do not call
+   * @code parent::__construct() @endcode.
+   */
+  public function __construct() {
+    $rules = module_invoke_all('fb_instant_articles_transformer_rules');
```

Comment:
Not sure we can get to where we want to be here given the constraints of the SDK.  I'm thinking one solution could be that `Drupal\fb_instant_articles\InstantArticle` simply stands alone, not extending anything but soaks up what might be common to `Drupal\fb_instant_articles_display\InstantArticle` and any other such child class, but otherwise stays the same to what it is in `7.x-1.x`.

And/or we move to a decorator pattern, which would implement all the public methods of `Drupal\fb_instant_articles\InstantArticle` and simply delegate to the object instance which is a member of the class for those methods, except for when we want to do something special, eg. for `render()`.

Although it's starting to feel like a lot of effort for something maybe not as crucial as we imagine it to be?